### ISSUE TITLE
linter: false negative for `no-unused-vars` for variable only used with `typeof`

### ISSUE THREAD

**Issue by rtarpine-vanta** [oxc-project/oxc#21536]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

`oxlint file.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": { "correctness": "off" },
  "rules": { "no-unused-vars": "error" },
}
```

### What happened?

When linting
```ts
const BOOLEANS = ["true", "false", "file-not-found"] as const;
export type Boolean = (typeof BOOLEANS)[number];
```
eslint reports that `'BOOLEANS' is assigned a value but only used as a type` when the `@typescript-eslint/no-unused-vars` rule is enabled. 

But oxlint does not report this violation.

**Comment by camc314** [comment]
Hi, 

we added [`reportVarsOnlyUsedAsTypes`](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-vars.html#reportvarsonlyusedastypes) option to oxlint as a result of user feedback.

This option defaults to `false`, but enabled with:
```
{ "rules": { "no-unused-vars": ["error", { "reportVarsOnlyUsedAsTypes": false }]}
```

Hopefully this works for you

**Comment by camc314** [comment]
more context:

https://github.com/oxc-project/oxc/pull/11009

https://github.com/oxc-project/oxc/issues/11006

**Comment by rtarpine-vanta** [comment]
Ok, thanks for letting me know! We'll probably use that setting to keep things stricter like eslint.

But it's interesting to contrast this with https://github.com/oxc-project/oxc/issues/20441, where oxlint intentionally treats type guard parameters as unused when only used in the type predicate.

**Comment by camc314** [comment]
If it wasn't a breaking change, I'd like to go back and at least change the default option of `reportVarsOnlyUsedAsTypes` to be more strict.

Iirc, originally the problem was that so many people were defining types with zod schemas, and using `z.infer` to get the type of the schema, but they never actually used the schema.
